byterun/pyvm2.py

```diff
@@ -166,18 +167,36 @@ def unwind_block(self, block):
 
     def parse_byte_and_args(self):
         """ Parse 1 - 3 bytes of bytecode into
-        an instruction and optionally arguments."""
+        an instruction and optionally arguments.
+        In Python3.6 the format is 2 bytes per instruction."""
         f = self.frame
         opoffset = f.f_lasti
-        byteCode = byteint(f.f_code.co_code[opoffset])
+        if f.py36_opcodes:
+            currentOp = f.py36_opcodes[opoffset]
+            byteCode = currentOp.opcode
+            byteName = currentOp.opname
+        else:
+            byteCode = byteint(f.f_code.co_code[opoffset])
+            byteName = dis.opname[byteCode]
         f.f_lasti += 1
-        byteName = dis.opname[byteCode]
         arg = None
         arguments = []
+        if f.py36_opcodes and byteCode == dis.EXTENDED_ARG:
```

Comment:
I don't think extended_arg is new to 3.6, can we remove the first part of this check?

Reply:
sure, I will update that.

Reply:
As I mentioned previously, this approach only works for Python 3.6 and above, and only if we are using the `get_instructions` api.